src/server/services/saml.strategy.ts

```diff
@@ -24,22 +24,24 @@ class SAMLStrategy extends PassportStrategy(Strategy) {
 
   public async validate(profile?: HarvardKeyProfile): Promise<User> {
     if (this.devMode) {
-      return new User({
+      const dummyUser: User = {
         id: 'abc123',
         firstName: 'Test',
         lastName: 'User',
         email: 'noreply@seas.harvard.edu',
-      });
+      };
+      return dummyUser;
     }
     if (!profile) {
       throw new UnauthorizedException('You are not authorized to use this application. Please contact SEAS computing');
     }
-    return new User({
+    const authenticatedUser: User = {
```

Comment:
So what's the difference between declaring these as `User`s with types vs. instantiating them with a constructor? Is that just necessary for TypeORM, or is there some other benefit?

Reply:
It's for TypeORM, yeah. TypeORM (I think) instantiates them internally by doing `new Foo()` and then setting properties. So when it does that, no values are passed to the constructor, causing the constructor to throw.